src/TS/services/api_calls.ts

```diff
@@ -1,50 +1,120 @@
 import API_BASE_URL from '../api/api_Base_Url';
 import { KartRun } from '../models/KartRun';
 import { LapDetails } from '../models/lapDetials';
+import {
+  Subject,
+  switchMap,
+  map,
+  catchError,
+  EMPTY,
+  from,
+  mergeMap,
+} from 'rxjs';
+import { fromFetch } from 'rxjs/fetch';
+import { fromPromise } from 'rxjs/internal/observable/innerFrom';
 
-export function getFileName(
-  callBack: (s: string[]) => void,
-  errorCallBack: (e: Error) => void,
-) {
-  fetch(`${API_BASE_URL}/runs`)
-    .then((response) => response.json())
-    .then((responseJson: string[]) => {
-      callBack(responseJson);
-    })
-    .catch((err) => errorCallBack(err));
-}
+export const newFetchRequest$ = new Subject<null>();
```

Comment:
Maybe try void instead of null. Just a suggestion, null is fine too.

Reply:
I will take that into consideration, thank you.